### PR TITLE
link back to dashboard with params

### PR DESCRIPTION
**Pull request by sandikbarr**

### CODE REVIEW

File: src/app/campaign-list/campaign-list.module.ts [hunk 1]
```diff
@@ -53,9 +53,6 @@ import { CampaignListService } from './campaign-list.service';
   ],
   exports: [
     CampaignListComponent
-  ],
```
Comment (sandikbarr) [inline src/app/campaign-list/campaign-list.module.ts:56]: CampaignListService is now `providedIn: 'root'` as a [singleton](https://angular.io/guide/singleton-services)

File: src/app/campaign-list/campaign-list.service.mock.ts [hunk 2]
```diff
@@ -70,12 +68,12 @@ export const campaigns: Campaign[] = [
...
-  type: [{id: 'house', label: 'House'}, {id: 'paid_campaign', label: 'Paid Campaign'}],
-  geo: [{id: 'US', label: 'United States'}, {id: 'CA', label: 'Canada'}],
-  zone: [{id: 'mid_1', label: 'Midroll'}, {id: 'pre_1', label: 'Preroll'}]
+  advertiser: [{ id: 3, label: 'Toyota' }, { id: 2, label: 'Griddy' }, { id: 1, label: 'Adidas' }],
+  podcast: [{ id: 3, label: 'Podcast 3' }, { id: 2, label: 'Podcast 2' }, { id: 1, label: 'Podcast 1' }],
+  status: [{ id: 'canceled', label: 'Canceled' }, { id: 'approved', label: 'Approved' }, { id: 'sold', label: 'Sold' }],
+  type: [{ id: 'house', label: 'House' }, { id: 'paid_campaign', label: 'Paid Campaign' }],
+  geo: [{ id: 'US', label: 'United States' }, { id: 'CA', label: 'Canada' }],
+  zone: [{ id: 'mid_1', label: 'Midroll' }, { id: 'pre_1', label: 'Preroll' }]
 };
```
Comment (sandikbarr) [inline src/app/campaign-list/campaign-list.service.mock.ts:77]: (_sorry about all the prettier whitespace changes_)

File: src/app/campaign-list/campaign-list.service.ts [hunk 3]
```diff
@@ -29,6 +29,15 @@ export interface CampaignParams {
   desc?: boolean;
 }
 
+// TODO: Omit is added in TS 3.5
```
Comment (sandikbarr) [inline src/app/campaign-list/campaign-list.service.ts:32]: [Omit](https://www.typescriptlang.org/docs/handbook/utility-types.html#omittk) comes in TypeScript 3.5. I tried to install it, but Angular 7 isn't ready for it.
Omit is useful so I can show the association between the CampaignParams and the CampaignRouteParams interfaces; the properties that are overridden in route params are string interpretations of the values.

File: src/app/campaign-list/campaign-list.service.ts [hunk 4]
```diff
@@ -59,116 +68,121 @@ export interface Campaign {
...
-          return {...acc, [doc.id]: {loading: true}};
-        }, {}));
-        // campaigns state is mapped from a list of ids from the current request/filter params
-        this._currentCampaignIds = campaignDocs.map(c => c['id']);
-        return campaignDocs.map(doc => {
-          return combineLatest(
-            of(doc),
-            doc.follow('prx:advertiser'),
-            doc.followItems('prx:flights')
+    const filters = this.getFilters({ advertiser, podcast, status, type, geo, zone, text, representative, before, after });
```
Comment (sandikbarr) [inline src/app/campaign-list/campaign-list.service.ts:117]: (_big prettier whitespace change_)

File: src/app/campaign-list/campaign-list.service.ts [hunk 5]
```diff
@@ -232,4 +246,60 @@ export class CampaignListService {
     }
     return filters;
   }
+
```
Comment (sandikbarr) [inline src/app/campaign-list/campaign-list.service.ts:249]: I pulled this code from the campaign-list.component to re-use it for building the queryParams

File: src/app/campaign-list/campaign-list.module.ts (hunk 1 above)
Comment (cavis) [reply src/app/campaign-list/campaign-list.module.ts:56]: +1

File: src/app/campaign-list/campaign-list.service.ts (hunk 3 above)
Comment (cavis) [reply src/app/campaign-list/campaign-list.service.ts:32]: Oooh, that is neat!

Comment (cavis) [reply src/app/campaign-list/campaign-list.service.ts:32]: And the other utility types in there are also great

**Review by cavis — APPROVED** [review]
:+100: as advertised!  (And linting ++)

**Pull request merged**